File: 153-find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        n = len(nums)

        first = 0
        last = n - 1

        while first < last:
            mid = (first + last) // 2

            if nums[mid] > nums[last]:
                first = mid + 1

            else:
                last = mid

        return nums[first]

File: 153-find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMin = function (arr) {
    let first = 0, last = arr.length - 1;

    while (first < last) {
        let mid = Math.floor((first + last) / 2);

        if (arr[mid] > arr[last]) {
            first = mid + 1;
        }
        else {
            last = mid;
        }
    }

    return arr[first];
};